src/Raven.Server/Documents/Handlers/TimeSeriesHandler.cs

```diff
@@ -411,118 +411,80 @@ public async Task ConfigTimeSeries()
                     {
-                        foreach (var kvp in _appendDictionary)
+                        var deletionRange = new TimeSeriesStorage.DeletionRangeRequest
                         {
-                            LastChangeVector = tss.AppendTimestamp(context,
-                                operation.DocumentId,
-                                docCollection,
-                                kvp.Key,
-                                kvp.Value.Values
-                            );
-
-                            changes += kvp.Value.Values.Count;
-                        }
+                            DocumentId = _operation.DocumentId,
+                            Collection = docCollection,
+                            Name = _operation.Name,
+                            From = removal.From,
+                            To = removal.To
+                        };
 
-                        _appendDictionary.Clear();
-                    }
-                    else if (_singleValue != null)
-                    {
-                        LastChangeVector = tss.AppendTimestamp(context,
-                            operation.DocumentId,
-                            docCollection,
-                            _singleValue.Name,
-                            new[]
-                            {
-                            _singleValue
-                            });
+                        LastChangeVector = tss.RemoveTimestampRange(context, deletionRange);
 
                         changes++;
                     }
-
                 }
 
-                return changes;
-            }
-
-            private void ConvertBatch(TimeSeriesOperation operation)
-            {
-                _appendDictionary?.Clear();
-                _singleValue = null;
+                if (_operation.Appends?.Count > 0 == false)
+                    return changes;
 
-                if (operation.Appends == null || operation.Appends.Count == 0)
+                if (_operation.Appends.Count == 1)
                 {
-                    return;
-                }
+                    LastChangeVector = tss.AppendTimestamp(context,
+                        _operation.DocumentId,
+                        docCollection,
+                        _operation.Name,
+                        new[] {_operation.Appends[0]});
 
-                if (operation.Appends.Count == 1)
-                {
-                    _singleValue = operation.Appends[0];
-                    return;
+                    changes++;
                 }
-
-                _appendDictionary = new Dictionary<string, SortedList<long, TimeSeriesOperation.AppendOperation>>();
-
-                foreach (var item in operation.Appends)
+                else
                 {
-                    if (_appendDictionary.TryGetValue(item.Name, out var sorted) == false)
+                    var sorted = new SortedList<long, TimeSeriesOperation.AppendOperation>();
```

Comment:
When I think about it, in most cases it is redundant, would it be better to assume that we get a sorted data from the client.

Otherwise we will throw during the enumeration?

Reply:
@ayende @ppekrol

Reply:
there is also the problem of having repeating timestamps in `_operation.Appends` 
(like in `CanCreateSimpleTimeSeries2` test) - which this code solves